android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt

```diff
@@ -273,22 +297,23 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
 
         val trackList = Arguments.toList(data)
         val queue = musicService.tracks
-        val indexes: MutableList<Int> = ArrayList()
-        for (o in trackList!!) {
-            val index = if (o is Int) o else o.toString().toInt()
-
-            // We do not allow removal of the current item
-            musicService.getCurrentTrackIndex {
-                if (index == it) return@getCurrentTrackIndex
-                if (index >= 0 && index < queue.size) {
-                    indexes.add(index)
+
+        if (trackList != null) {
+            for (track in trackList) {
+                val index = if (track is Int) track else track.toString().toInt()
+
+                // We do not allow removal of the current item
+                musicService.getCurrentTrackIndex {
```

Comment:
Hm.. should we really be calling this async method over and over again then? Maybe its best to just get the current index once then do all comparisons against that?

since we can't await or suspend maybe something like:
```kotlin
if (trackList != null) {
  musicService.getCurrentTrackIndex {
    val indexes: MutableList<Int> = ArrayList()
    for (track in trackList) {
      val index = if (o is Int) o else o.toString().toInt()
      if (index == it) return@getCurrentTrackIndex
        if (index >= 0 && index < queue.size) {
          indexes.add(index)
    }
    
    if (indexes.isNotEmpty())
      musicService.remove(indexes)
    }
}
```

_This was written in the browser so might not be perfect.. but you get the idea_

Reply:
Yeah sure, this is a nice optimization :)

Reply:
@dcvz try now